src/daft-core/src/array/ops/utf8.rs

```diff
@@ -271,6 +270,30 @@ fn replace_on_literal<'a>(
     Ok(Utf8Array::from((name, Box::new(arrow_result?))))
 }
 
+fn substring(s: &str, start: usize, len: Option<usize>) -> Option<&str> {
+    let mut char_indices = s.char_indices().skip(start);
+    let len = match len {
+        Some(len) => {
+            if len == 0 {
+                return None;
+            } else {
+                len
+            }
+        }
+        None => s.len(),
+    };
```

Comment:
A small performance improvement could be to just return `&s[start_pos..]` if `len` is `None`, that way we don't need to iterate further on `char_indices`